src/query/api/v1/handler/prometheus/common.go

```diff
@@ -111,8 +111,18 @@ func ParseRequestTimeout(
 	configFetchTimeout time.Duration,
 ) (time.Duration, error) {
 	timeout := r.Header.Get("timeout")
-	if timeout == "" {
-		return configFetchTimeout, nil
+	if len(timeout) == 0 {
+		if r.Body != nil {
+			if err := r.ParseForm(); err != nil {
+				return 0, err
+			}
+
+			timeout = r.FormValue("timeout")
```

Comment:
This is safe even when timeout is absent I assume?

Reply:
It'll be an empty string, yeah